engine/src/Interpreter/interpreter_ops.cuh

```diff
@@ -447,13 +447,22 @@ private:
 					get_data_from_buffer(&right_value, buffer, right_position);
 				}
 				right_valid = getColumnValid(row_valids, right_position);
+
+				if(oper == operator_type::BLZ_DIV && right_value == 0){ //if div by zero = null
+					right_valid = false;
+				}
 			}	else if(right_position == SCALAR_INDEX) {
 				if (is_string_type(right_type_id)) {
 					right_str_view = static_cast<cudf::string_scalar_device_view*>(scalars_right[op_index])->value();
+					right_valid = true;
 				} else {
 					right_value = get_scalar_value<RightType>(scalars_right[op_index]);
+
+					if(oper == operator_type::BLZ_DIV && right_value == 0) //if div by zero = null
+						right_valid = false;
+					else
+						right_valid = true;
```

Comment:
Maybe move this logic to be just in one place in line 598
https://github.com/BlazingDB/blazingsql/blob/ad55fb1207838917e356e128513e955047f2a675/engine/src/Interpreter/interpreter_ops.cuh#L597-L599

Reply:
@jeanp413 I moved the logic. Please, check it again.